### PR TITLE
rename QuickerSort to ScratchQuickSort

### PR DESCRIPTION
**Pull request by LilithHafner**
From https://github.com/JuliaLang/julia/pull/47973#issuecomment-1373884035 and discussion on slack.

fwiw, the term we use in sorting is "scratch space" and the (undocumented) keyword argument is named "scratch".

### CODE REVIEW

**Review by StefanKarpinski — APPROVED** [review]

**Comment by LilithHafner** [comment]
I actually prefer `QuickScratchSort`. It reflects the strong similarity with `QuickSort`, reflects the out of place nature in a neutral way, and uses terminology consistent with the keyword argument name `scratch`.

@petvana, I'm curious what your objection to renaming `QuickerSort` => `QuickBufferSort` is and whether it still applies to `QuickScratchSort`.

**Comment by petvana** [comment]
I'm not a native speaker. But `QuickBufferSort` sounds like an improved version of `BufferSort` (and it is not). Moreover, whether in-place or stability is a more important feature is questionable. I'd use `StableQuickSort` as it seems to be already used in literature; the only problem is the name collision. May an alternative be `BufferQuickSort` or `AllocatingQuickSort`? You may ignore my comment if you see it as irrelevant.

**Comment by LilithHafner** [comment]
Thank you! The perspective of a non-native speaker is important to me because I want this name to make sense to non-native speaker users.

`StableQuickSort` is not ideal because this algorithm could be adapted to a slightly faster unstable algorithm by deleting [this line](https://github.com/JuliaLang/julia/blob/de73c26fbff61d07a38c9653525b530a56630831/base/sort.jl#L1070) and a few others. I don't like this possible future: `sort(v; alg=StableQuickSort, stable=false)`.

`AllocatingQuickSort` is not correct because this algorithm need not allocate if passed a preallocated scratch space:

```julia
julia> @btime sort!(rand!(x); alg=QuickSort) setup=(x=rand(Int, 1000));
  33.422 μs (0 allocations: 0 bytes)

julia> @btime sort!(rand!(x); alg=Base.Sort.QuickerSort(), scratch) setup=(x=rand(Int, 1000); scratch=rand(Int, 1000));
  21.655 μs (0 allocations: 0 bytes)

julia> VERSION
v"1.9.0-beta2"
```

**Perhaps `ScratchQuickSort`?** This no longer implies that it is a quick variant of `ScratchSort` (does not exist) and also uses the same terminology as the keyword argument.

**Comment by petvana** [comment]
+1 for `ScratchQuickSort`

File: base/sort.jl [hunk 1]
```diff
@@ -953,12 +953,12 @@ end
...
-    QuickerSort(lo::Union{Integer, Missing}, hi::Union{Integer, Missing}=lo, next::Algorithm=SMALL_ALGORITHM) <: Algorithm
+    ScratchQuickSort(next::Algorithm=SMALL_ALGORITHM) <: Algorithm
+    ScratchQuickSort(lo::Union{Integer, Missing}, hi::Union{Integer, Missing}=lo, next::Algorithm=SMALL_ALGORITHM) <: Algorithm
 
-Use the `QuickerSort` algorithm with the `next` algorithm as a base case.
+Use the `ScratchQuickSort` algorithm with the `next` algorithm as a base case.
 
-`QuickerSort` is like `QuickSort`, but utilizes scratch space to operate faster and allow
+`ScratchQuickSort` is like `QuickSort`, but utilizes scratch space to operate faster and allow
 for the possibility of maintaining stability.
```
Comment (LilithHafner) [inline base/sort.jl:962]: This feels right

**Pull request merged**